### PR TITLE
feat: add support for RISCV32 toolchains

### LINKED ISSUE #7: Add support for Rust
Hi, I'd like to build code for ESP32 in Rust, on my NixOS machine. I though this would be a good start. Unfortunately, this does not work yet. When I tried the following:

```console
https://github.com/espressif/rust-esp32-example.git
cd rust-esp32-example
idf.py set-target esp32
idf.py build
```
which errors with 

```console
  process didn't exit successfully: `rustc - --crate-name ___ --print=file-names --target xtensa-esp32-espidf --crate-type bin --crate-type rlib --crate-type dylib --crate-type cdylib --crate-type staticlib --crate-type proc-macro --print=sysroot --print=cfg` (exit status: 1)
  --- stderr
  error: Error loading target specification: Could not find specification for target "xtensa-esp32-espidf". Run `rustc --print target-list` for a list of built-in targets
```

This is expected - I did not install the Rust fork yet. However, and this is where I think we might have a problem in the nixpkgs-esp-dev project:

```console
https://github.com/espressif/rust-esp32-example.git
cd rust-esp32-example
idf.py set-target esp32c3
idf.py build
```

results in 

```console
-- The C compiler identification is unknown
-- The CXX compiler identification is unknown
-- The ASM compiler identification is unknown
-- Found assembler: riscv32-esp-elf-gcc
CMake Error at /nix/store/zxyj5pafs2v9dbc1ij30zxg00cyi20a9-esp-idf-v4.3.1/tools/cmake/project.cmake:311 (__project):
  The CMAKE_C_COMPILER:

    riscv32-esp-elf-gcc

  is not a full path and was not found in the PATH.

  Tell CMake where to find the compiler by setting either the environment
  variable "CC" or the CMake cache entry CMAKE_C_COMPILER to the full path to
  the compiler, or to the compiler name if it is in the PATH.
Call Stack (most recent call first):
  CMakeLists.txt:6 (project)


CMake Error at /nix/store/zxyj5pafs2v9dbc1ij30zxg00cyi20a9-esp-idf-v4.3.1/tools/cmake/project.cmake:311 (__project):
  The CMAKE_CXX_COMPILER:

    riscv32-esp-elf-g++

  is not a full path and was not found in the PATH.

  Tell CMake where to find the compiler by setting either the environment
  variable "CXX" or the CMake cache entry CMAKE_CXX_COMPILER to the full path
  to the compiler, or to the compiler name if it is in the PATH.
Call Stack (most recent call first):
  CMakeLists.txt:6 (project)


CMake Error at /nix/store/zxyj5pafs2v9dbc1ij30zxg00cyi20a9-esp-idf-v4.3.1/tools/cmake/project.cmake:311 (__project):
  The CMAKE_ASM_COMPILER:

    riscv32-esp-elf-gcc

  is not a full path and was not found in the PATH.

  Tell CMake where to find the compiler by setting either the environment
  variable "ASM" or the CMake cache entry CMAKE_ASM_COMPILER to the full path
  to the compiler, or to the compiler name if it is in the PATH.
Call Stack (most recent call first):
  CMakeLists.txt:6 (project)


-- Warning: Did not find file Compiler/-ASM
-- Configuring incomplete, errors occurred!
```

Is this nix version of the toolchain supposed to already work with the esp32c3 (which needs a RISC-V toolchain)?

### PR DESCRIPTION
**Pull request by RaitoBezarius**
Hi there, here's a PR that fixes #7 : I do not know what is the optimal way to handle this, rather, maybe in the esp32 toolchain rather than a new toolchain at all.

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -15,6 +15,7 @@
     {
       packages = {
         inherit (pkgs)
+          gccriscv32-esp32c3-elf-bin
```
Comment (mirrexagon) [inline flake.nix:18]: This is missing a `-`, should be `gcc-riscv32-esp32c3-elf-bin`.

**Review by mirrexagon — CHANGES_REQUESTED** [review]
Just a typo, but otherwise looks great! Thanks for doing this work.

**Comment by mirrexagon** [comment]
I think this is the best way to handle it right now. I don't want to combine toolchains as that would force users to download all toolchains even if they want to use just one.

In the future, if this toolchain is used for all ESP RISC-V microcontrollers, then I will probably rename the toolchain to be more generic - but for now I don't know how that will work so I think naming it specifically for the only current ESP RISC-V chip is best.

File: flake.nix (hunk 1 above)
Comment (RaitoBezarius) [reply flake.nix]: Indeed!

**Comment by RaitoBezarius** [comment]
> I think this is the best way to handle it right now. I don't want to combine toolchains as that would force users to download all toolchains even if they want to use just one.
> 
> In the future, if this toolchain is used for all ESP RISC-V microcontrollers, then I will probably rename the toolchain to be more generic - but for now I don't know how that will work so I think naming it specifically for the only current ESP RISC-V chip is best (ESP32-C3 as you have done).

Awesome :), addressed the changes! To be fair, we have been testing this in our hackerspace, and it seems to work fine on our ESP32-C3, we have been able to run some examples and our own code.

**Comment by mirrexagon** [comment]
That's awesome! Thank you for adding this!

**Pull request merged**